cni/pkg/install/cniconfig.go

```diff
@@ -26,113 +26,47 @@ import (
 	"github.com/containernetworking/cni/libcni"
 
 	"istio.io/istio/cni/pkg/config"
+	"istio.io/istio/cni/pkg/plugin"
 	"istio.io/istio/cni/pkg/util"
 	"istio.io/istio/pkg/file"
 )
 
-type pluginConfig struct {
-	mountedCNINetDir string
-	cniConfName      string
-	chainedCNIPlugin bool
-}
-
-type cniConfigTemplate struct {
-	cniNetworkConfigFile string
-	cniNetworkConfig     string
-}
-
-type cniConfigVars struct {
-	cniNetDir          string
-	kubeconfigFilename string
-	logLevel           string
-	k8sServiceHost     string
-	k8sServicePort     string
-	k8sNodeName        string
-	logUDSAddress      string
-	cniEventAddress    string
-}
-
-func getPluginConfig(cfg *config.InstallConfig) pluginConfig {
-	return pluginConfig{
-		mountedCNINetDir: cfg.MountedCNINetDir,
-		cniConfName:      cfg.CNIConfName,
-		chainedCNIPlugin: cfg.ChainedCNIPlugin,
-	}
-}
-
-func getCNIConfigTemplate(cfg *config.InstallConfig) cniConfigTemplate {
-	return cniConfigTemplate{
-		cniNetworkConfigFile: cfg.CNINetworkConfigFile,
-		cniNetworkConfig:     cfg.CNINetworkConfig,
+func createCNIConfigFile(ctx context.Context, cfg *config.InstallConfig) (string, error) {
+	pluginConfig := plugin.Config{
+		LogLevel:        cfg.LogLevel,
+		LogUDSAddress:   cfg.LogUDSAddress,
+		CNIEventAddress: cfg.CNIEventAddress,
+		AmbientEnabled:  cfg.AmbientEnabled,
+		Kubernetes: plugin.Kubernetes{
+			Kubeconfig:        filepath.Join(cfg.CNINetDir, cfg.KubeconfigFilename),
+			ExcludeNamespaces: strings.Split(cfg.ExcludeNamespaces, ","),
+		},
 	}
-}
 
-func getCNIConfigVars(cfg *config.InstallConfig) cniConfigVars {
-	return cniConfigVars{
-		cniNetDir:          cfg.CNINetDir,
-		kubeconfigFilename: cfg.KubeconfigFilename,
-		logLevel:           cfg.LogLevel,
-		k8sServiceHost:     cfg.K8sServiceHost,
-		k8sServicePort:     cfg.K8sServicePort,
-		k8sNodeName:        cfg.K8sNodeName,
-		logUDSAddress:      cfg.LogUDSAddress,
-		cniEventAddress:    cfg.CNIEventAddress,
-	}
-}
+	pluginConfig.Name = "istio-cni"
+	pluginConfig.Type = "istio-cni"
+	pluginConfig.CNIVersion = "0.3.1"
```

Comment:
include in the literal?

Reply:
Because the struct is nested with the "official" CNI library struct, it's a bit more verbose to do it that way.